src/main/java/seedu/application/logic/parser/FindCommandParser.java

```diff
@@ -21,34 +29,49 @@ public class FindCommandParser implements Parser<FindCommand> {
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
-        }
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_ROLE, PREFIX_COMPANY,
+                        PREFIX_DEADLINE, PREFIX_STATUS, PREFIX_JOB_TYPE, PREFIX_INDUSTRY);
 
-        String[] specifierAndKeywords = trimmedArgs.split("\\s+");
-        List<String> keywords = getKeywordList(specifierAndKeywords);
-        String specifier = specifierAndKeywords[0];
-        if (!(FieldContainsKeywordsPredicate.isValidSpecifier(specifier))) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_INVALID_SPECIFIER));
-        }
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_ROLE, PREFIX_COMPANY,
+                PREFIX_DEADLINE, PREFIX_STATUS, PREFIX_JOB_TYPE, PREFIX_INDUSTRY);
 
-        return new FindCommand(new FieldContainsKeywordsPredicate(specifier, keywords));
-    }
 
-    /**
-     * Takes the given array of which contains the specifier and the keywords
-     * and returns a list of keywords.
-     */
-    private List<String> getKeywordList(String[] specifierAndKeywords) throws ParseException {
-        ArrayList<String> keywords = new ArrayList<>();
-        for (int i = 1; i < specifierAndKeywords.length; i++) {
-            keywords.add(specifierAndKeywords[i]);
+
+        ArrayList<FieldContainsKeywordsPredicate> predicateList = new ArrayList<>();
+
+        for (Map.Entry<Prefix, List<String>> entry : argMultimap.getArgMultimap().entrySet()) {
+            if (entry.getKey().equals(new Prefix(""))) {
+                continue;
+            }
+
+            if (!FieldContainsKeywordsPredicate.isValidPrefix(entry.getKey())) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, Messages.MESSAGE_INVALID_PREFIX));
+            }
+
+            String keywords = entry.getValue().get(0);
+            if (keywords.equals("")) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_EMPTY_KEYWORDS));
+            }
+
+            String[] splitKeywords = keywords.split("\\s+");
+            List<String> keywordList = new ArrayList<>();
+            for (String keyword : splitKeywords) {
+                keywordList.add(keyword);
+            }
+
+            FieldContainsKeywordsPredicate predicate = new FieldContainsKeywordsPredicate(entry.getKey(), keywordList);
+            predicateList.add(predicate);
+        }
+
+        if (argMultimap.size() < 2) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        return keywords;
+        return new FindCommand(new CombinedPredicate(predicateList));
```

Comment:
Good use of another class `CombinedPredicate` to handle the operation with the `predicateList`